discovery/consul/consul.go

```diff
@@ -47,6 +48,25 @@ func NewConsulConfig(config interface{}) (*Consul, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	maxRetry := 30
+	retry := 0
+	// we need to wait for Consul to start and self-elect
```

Comment:
Hahaha you give me too much credit @tgross, I'm much lazier than this 😄. I went to go add a `WaitForLeader()` type method to our tests, and figured why not add it to the code that actually creates the client. I'm in full agreement with you, this will be gone in a moment, and I'd love to work on #164 in a separate PR as I have a similar desire/use case on wanting an app on startup to crash, but if it's already up, to stay running.